Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/AbstractSqlEamDb.java

```diff
@@ -1150,6 +1153,41 @@ public void setArtifactInstanceKnownStatus(EamArtifact eamArtifact, TskData.File
             EamDbUtil.closeConnection(conn);
         }
     }
+    
+    /**
+     * Set knownBad status for all files/artifacts in the given case that
+     * are tagged with the given tag name. 
+     * Files/artifacts that are not already in the database will be added.
+     * @param tagName The name of the tag to search for
+     * @param curCase The case to search in
+     */
+    @Override
+    public void setArtifactsKnownBadByTag(String tagNameString, Case curCase) throws EamDbException{
+        try{
+            TagName tagName = curCase.getServices().getTagsManager().getDisplayNamesToTagNamesMap().get(tagNameString);
+            
+            // First find any matching artifacts
+            List<BlackboardArtifactTag> artifactTags = curCase.getSleuthkitCase().getBlackboardArtifactTagsByTagName(tagName);                  
+            
+            for(BlackboardArtifactTag bbTag:artifactTags){
+                List<EamArtifact> convertedArtifacts = EamArtifactUtil.fromBlackboardArtifact(bbTag.getArtifact(), true, getCorrelationTypes(), true);
+                for (EamArtifact eamArtifact : convertedArtifacts) {
+                    setArtifactInstanceKnownStatus(eamArtifact,TskData.FileKnown.BAD);
+                }
+            }
+
+            // Now search for files
+            List<ContentTag> fileTags = curCase.getSleuthkitCase().getContentTagsByTagName(tagName);
+            for(ContentTag contentTag:fileTags){
+                final EamArtifact eamArtifact = EamArtifactUtil.getEamArtifactFromContent(contentTag.getContent(), 
+                            TskData.FileKnown.BAD, "");
+                setArtifactInstanceKnownStatus(eamArtifact, TskData.FileKnown.BAD);
+            }
+        } catch (TskCoreException ex){
+            throw new EamDbException("Error updating artifacts", ex);
+        }
+        
+    }
```

Comment:
I don't think this is the best place to put this method. My thinking is that adding methods to SleuthkitCase for lots of "special case" options is what has lead to a very large and therefore difficult to use SleuthkitCase API.  I would prefer to see a low-level service class such as the CR database provide primitive operations that can then be combined in various ways by clients to get the desired result. In my opinion, that this is not a primitive operation on the CR database is evident in the call to SleuthkitCase, which couples the CR database implementation to the case database implementation. I propose that we put this code in the ManageTagsDialog class instead. Then it will transfer nicely to the Tags options panel, which will then act as a client of the low level case database and CR database classes, linking them together to get create an operation specific to the UI.